### ISSUE TITLE
Change color palette component markup to fieldset with radio buttons

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#16105]
**Is your feature request related to a problem? Please describe.**
Related to #15307. The color palette component in the Paragraph color settings has a series of buttons of which only one can be chosen at a time; the buttons aren't grouped and there is no accessible labelling for them. 

**Describe the solution you'd like**
The markup could be changed to a fieldset with legend wrapping a group of radio buttons for the colors and maybe a couple of regular buttons for the "Custom color" and "Clear" options. Feedback welcome on this as I'm not sure if there's an issue with grouping radio and other buttons in the same fieldset.

**Describe alternatives you've considered**
Even if we don't go with radio buttons this component might benefit from having a fieldset as wrapper to indicate the relationship between all its buttons.

**Comment by afercia** [comment]
@tellthemachines thanks for your report!

> there is no accessible labelling for them.

Worth noting the buttons are labelled with an `aria-label` attribute, e.g.:
```
<button type="button" aria-label="Color: Pale pink" ...
```

Also, #15307 already proposes to group these buttons within a fieldset + legend element. Maybe I'm missing something but seems to me what you're proposing is already covered in #15307 🙂 Please do feel free to report uour considerations on #15307, which is absolutely open to discussion.

If I missed something, please do feel free to reopen this ticket, thanks!
